orc/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -531,6 +543,34 @@ public void nonNullWrite(int rowId, Map<K, V> map, ColumnVector output) {
     }
   }
 
+  private static class DeleteWriter implements OrcRowWriter<PositionDelete<Record>> {
```

Comment:
I think the naming has to be a bit more specific as this is a position delete.
Also, it should probably accept a path transformation function.

Reply:
Renamed the class to `PositionDeleteWriter`.
Let's discuss the path transformation function in the thread above.